saleor/graphql/order/bulk_mutations/orders.py

```diff
@@ -0,0 +1,47 @@
+import graphene
+
+from ....order import OrderEvents, models
+from ....order.utils import cancel_order
+from ...core.mutations import BaseBulkMutation
+from ..mutations.orders import clean_order_cancel
+
+
+class OrdersCancel(BaseBulkMutation):
+    class Arguments:
+        ids = graphene.List(
+            graphene.ID,
+            required=True,
+            description='List of orders IDs to cancel.')
+        restock = graphene.Boolean(
+            required=True,
+            description='Determine if lines will be restocked or not.')
+
+    class Meta:
+        description = 'Cancels orders.'
+        model = models.Order
+
+    @classmethod
+    def clean_instance(cls, info, instance):
+        clean_order_cancel(instance)
+
+    @classmethod
+    def user_is_allowed(cls, user, input):
+        return user.has_perm('order.manage_orders')
+
+    @classmethod
+    def perform_mutation(cls, root, info, ids, **data):
+        data['user'] = info.context.user
+        return super().perform_mutation(root, info, ids, **data)
+
+    @classmethod
+    def bulk_action(cls, queryset, user, restock):
+        for order in queryset:
+            cancel_order(order=order, restock=restock)
+            if restock:
+                order.events.create(
+                    type=OrderEvents.FULFILLMENT_RESTOCKED_ITEMS.value,
+                    user=user,
+                    parameters={'quantity': order.get_total_quantity()})
```

Comment:
This code generating the event is repeated three times over the project, maybe we could do something or wait for https://github.com/mirumee/saleor/issues/2443#issuecomment-483205819.